src/group_theory/group_action/conj.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2021 . All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import group_theory.group_action.basic
+/-!
+# Conjugation action of a group on itself
+This file defines the conjugation action of a group on itself
+
+## Main definitions
+A type alias `conj G` is introduced for a group `G`. The group `conj G` acts on `G` by conjugation.
+
+-/
+
+variables (G : Type*) [group G]
+
+/-- A type alias for a group `G`. `conj G` acts on `G` by conjugation -/
+@[derive group] def conj : Type* := G
```

Comment:
I think it would be better to only require
```suggestion
variables (G : Type*) [div_inv_monoid G]

/-- A type alias for a group `G`. `conj G` acts on `G` by conjugation -/
@[derive div_inv_monoid] def conj : Type* := G
```
because then you can use this for `group_with_zero` too.

Reply:
The trouble with this is that it's not an action for a `div_inv_monoid` because you need `1⁻¹ = 1` and `(a * b)⁻¹ = b⁻¹ * a⁻¹` to prove it is an action and `div_inv_monoid` has no axioms about inv and div other than `a / b = a * b⁻¹`.

Reply:
Sure; you declare `has_scalar` for `div_inv_monoid`, and promote it to a `mul_action`  for `group` and `group_with_zero`.

Reply:
This comment still applies, I think

Reply:
Why would you want a `has_scalar` instance with no properties? It just causes confusion if someone uses it and then wonders why `one_smul` doesn't work and then wastes half an hour working out what's going on. If we add it, I wouldn't want anybody to try to use it, and for that reason I don't think we should add it, because someone might try to use it if we add it.

Reply:
My point here is the the conjugation in a group_with_zero is definitely worth having, and that by having a single `has_scalar` you don't need two different `smul_def` lemmas for each instance.

I don't think your concern is valid; I don't think there are any examples of `div_inv_monoid` that do not also satisfy `group` or `group_with_zero`; so in practice, it would be impossible to use the `has_scalar` instance without also having a `mul_action` available.

Reply:
I guess I could add an instance that says it's a sensible action in a `group_with_zero` and then it makes more sense.

Reply:
I guess it's also worth pointing out that I don't think this code should be used very much even for reasoning about conjugation. The main motivation is to be able to use a couple of group action lemmas, but for most of the time I think just using `mul_aut.conj` directly instead of these instance is the sensible approach.